apps/server/src/modules/files-storage-client/mapper/error.mapper.ts

```diff
@@ -1,23 +1,22 @@
 import { BadRequestException, ForbiddenException, InternalServerErrorException } from '@nestjs/common';
-import { ApiValidationError } from '@shared/common';
-import { FileStorageErrors, IFileStorageErrors } from '../interfaces';
+import { ErrorUtils } from '@src/core/error/utils';
+import { IError } from '@shared/infra/rabbitmq';
 
-export const isValidationError = (error: IFileStorageErrors): boolean => {
-	const checked = !!(error.validationErrors && error.validationErrors.length > 0);
+// type MapperError = BadRequestException | ForbiddenException | InternalServerErrorException;
 
-	return checked;
-};
 export class ErrorMapper {
-	static mapErrorToDomainError(errorObj: IFileStorageErrors): FileStorageErrors {
-		let error: FileStorageErrors;
-		if (errorObj.status === 400 && isValidationError(errorObj)) {
-			error = new ApiValidationError(errorObj.validationErrors);
-		} else if (errorObj.status === 400 && !isValidationError(errorObj)) {
+	static mapRpcErrorResponseToDomainError(
+		errorObj: IError
+	): BadRequestException | ForbiddenException | InternalServerErrorException {
+		let error: BadRequestException | ForbiddenException | InternalServerErrorException;
+		if (errorObj.status === 400) {
 			error = new BadRequestException(errorObj.message);
 		} else if (errorObj.status === 403) {
 			error = new ForbiddenException(errorObj.message);
+		} else if (errorObj.status === 500) {
+			error = new InternalServerErrorException(errorObj.message);
 		} else {
-			error = new InternalServerErrorException(errorObj);
+			error = new InternalServerErrorException(ErrorUtils.convertUnknownError(errorObj));
```

Comment:
Here you hand in the HttpExceptionOptions as first argument which is not correct. The first argument should be a string.